### PR TITLE
Update function to adjust tempBattle stats in saveData

### PR DESCRIPTION
**Pull request by CorgiOnNeptune**
Created a function for this as well as these temp battle issues are becoming a more common occurrence, so it's nice to have a quick and easy fix to toss in an update rather than editing individual affected files.

Not sure what broke the stats for the battles I included, could be script related, could not be. But, doesn't hurt to fix.

### CODE REVIEW

File: src/scripts/Update.ts [hunk 1]
```diff
@@ -2309,6 +2315,25 @@ class Update implements Saveable {
         }
     }
 
+    // Use setBattleState as 0 or 1 to manipulate battles to what status they should be based on related questline progress.
+    static fixTempBattleState = (saveData, battleName: string, setBattleState: number, questLineName: string, questStep: number) => {
+        const ql = saveData.quests.questLines.find((q) => q.name === questLineName);
```
Comment (RedSparr0w) [inline src/scripts/Update.ts:2320]: May be worth just chucking a return in here if no quest found?

File: src/scripts/Update.ts [hunk 2]
```diff
@@ -2309,6 +2315,25 @@ class Update implements Saveable {
...
+
+        if (setBattleState === 1) {
+            // set to complete if related questline/step is completed
+            if (ql?.state === 2 || ql?.quest > questStep) {
+                saveData.statistics.temporaryBattleDefeated[GameConstants.getTemporaryBattlesIndex(battleName)] = 1;
```
Comment (RedSparr0w) [inline src/scripts/Update.ts:2325]: Unfortunately we can't make use of the `getTemporaryBattlesIndex` as the indexes can change from update to update, so we need to use magic numbers (current index as of this update).

File: src/scripts/Update.ts [hunk 3]
```diff
@@ -2309,6 +2315,25 @@ class Update implements Saveable {
...
+            }
+        }
+
+        if (setBattleState === 0) {
+            // set to not complete if related questline/step isn't complete
+            if (ql?.state < 2 && ql?.quest <= questStep) {
+                saveData.statistics.temporaryBattleDefeated[GameConstants.getTemporaryBattlesIndex(battleName)] = 0;
```
Comment (RedSparr0w) [inline src/scripts/Update.ts:2332]: Same as above with magic numbers

Comment (CorgiOnNeptune) [reply src/scripts/Update.ts]: Changed battleNames to the index and added a return if `ql` is undefined

**Review by RedSparr0w — APPROVED** [review]
Changes look good, thanks for the PR! 👍

**Pull request merged**